op-deployer/pkg/deployer/bootstrap/implementations.go

```diff
@@ -70,6 +70,9 @@ func (c *ImplementationsConfig) Check() error {
 	if c.ArtifactsLocator == nil {
 		return errors.New("artifacts locator must be specified")
 	}
+	if c.ArtifactsLocator.IsTag() && c.ArtifactsLocator.Tag != c.L1ContractsRelease {
```

Comment:
We should set a sane default here in addition to the validation

Reply:
I gave the `--l1-contracts-release` flag a default value of `dev`. Should we just get rid of the flag altogether and just set `c.L1ContractsRelease` to either `dev` or `c.ArtifactsLocator.Tag` here instead?

Reply:
On second thought, I think it's probably overkill to do this validation. I think you can set the default to `dev` then in the docs explain what people can set it to otherwise.

Reply:
I was trying to abide by this guidance:
```
if they're deploying from one of our tagged releases, it should match the contracts locator
```
If that's a hard requirement, wouldn't this validation help remove a footgun?

Reply:
It doesn't matter very much. All this flag does is set a version string inside the OPCM. I guess the best UX would be to set it to the artifacts locator if it's a tag or`dev` otherwise, and remove the flag altogether like you suggest above.

Reply:
Removed the flag in https://github.com/ethereum-optimism/optimism/pull/15364/commits/fc6d013005b7e6cee9ba77549bafa03c61d8d7f4